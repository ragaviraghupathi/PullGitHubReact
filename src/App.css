tbody {
  height: 450px;
  width: 79%;
  overflow-y: auto;
  position: absolute;
}
tr {
  width: 100%;
  display: inline-table;
  table-layout: fixed;
}
.tablealign {
  text-align: center;
}
.error {
  color: red;
  padding-left: 15px;
}
