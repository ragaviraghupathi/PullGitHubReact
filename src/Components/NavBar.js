import React from 'react';
import {  DEFAULTUSER } from '../constants';

export default class NavBar extends React.Component {
   state = {
    searchUser: ''
  }
  
  handleChange = event => {
    this.setState({ searchUser: event.target.value });
  }
  
  handleSubmit = event => {
    event.preventDefault();
    let search= this.query.value.length !== 0 ? this.query.value :DEFAULTUSER ;
    this.props.onSearch(search);
    event.currentTarget.reset();
  }
  
  render() {  
    return (
      <div className="container">
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
        <h1 className="navbar-brand" >Github Repositories</h1>
        <form className="form-inline my-2 my-lg-0" style={{"float":"right"}} onSubmit={this.handleSubmit}>
            <input className="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search" onChange={this.handleChange} name="search"  ref={(input) => this.query = input}/>
           <button className="btn btn-outline-success my-2 my-sm-0" type="submit" style={{"width":"80px"}}>Search</button>
        </form>
        </nav>
      </div>
    );
  }
}