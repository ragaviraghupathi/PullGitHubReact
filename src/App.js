import React,{Component} from 'react';
import './App.css';
import axios from 'axios';
import { GITHUBURI,DEFAULTUSER } from './constants';
import NavBar from './Components/NavBar.js';
import RepositoriesList from './Components/RepositoriesList';
import { Error } from './Components/Error';

export default class App extends Component {
   constructor() {
     super();
     this.state = {
      repositories: [],
      loading: true,
      error:false
      };
    } 
     
  componentDidMount() {
    this.handleSearch();
  }
     
   handleSearch = (username = DEFAULTUSER) => {
    axios.get(GITHUBURI+`${username}/repos`)
      .then(response => {
        this.setState({
          username: username,
          repositories: response.data,
             loading: false,
          error:false
        });
      })
      .catch(error => {
        this.setState({ error: true });
      });    
  }
  
  render() { 
   const { username, repositories, error,loading}= this.state ;
    return (
      <div  className="container">
        <NavBar onSearch={this.handleSearch} />      
        {(error)
          ? <Error />
          : (loading)
            ? <p>Loading...</p>
            : <div className="container">
                <h2> user : {username}</h2>
                <RepositoriesList data={repositories} />
              </div>
          }          
      </div>
    );
  }
}
