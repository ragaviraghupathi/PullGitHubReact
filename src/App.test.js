import React from 'react';
import {render , fireEvent} from '@testing-library/react';
import App from './App';
import NavBar from './Components/NavBar';

test('renders header', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Github Repositories/);
  expect(linkElement).toBeInTheDocument();
});

test('render to match App Component', () => {
  const { asFragment } = render(<App/>);
  expect(asFragment()).toMatchSnapshot();
});

test('test the search functionality', () => {
  const { getByText, asFragment } = render(<NavBar onSearch={() => { }} />)
  const firstRender = asFragment()
  fireEvent.click(getByText(/Search/))
  expect(firstRender).toMatchSnapshot(asFragment())
});





