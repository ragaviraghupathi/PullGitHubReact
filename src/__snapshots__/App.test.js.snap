// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render to match App Component 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <nav
        class="navbar navbar-expand-lg navbar-dark bg-dark"
      >
        <h1
          class="navbar-brand"
        >
          Github Repositories
        </h1>
        <form
          class="form-inline my-2 my-lg-0"
          style="float: right;"
        >
          <input
            aria-label="Search"
            class="form-control mr-sm-2"
            name="search"
            placeholder="Search"
            type="search"
          />
          <button
            class="btn btn-outline-success my-2 my-sm-0"
            style="width: 80px;"
            type="submit"
          >
            Search
          </button>
        </form>
      </nav>
    </div>
    <p>
      Loading...
    </p>
  </div>
</DocumentFragment>
`;

exports[`test the search functionality 1`] = `Object {}`;
